internal/backends/sqlite/metadata/pool/pool.go

```diff
@@ -244,3 +244,34 @@ func (p *Pool) Drop(ctx context.Context, name string) bool {
 
 	return true
 }
+
+// InTransaction wraps the given function f in a SQLite transaction.
+//
+// If f returns an error, the transaction is rolled back.
+//
+// Passed context will be used for transaction.
+// Context cancellation DOES rollback the transaction.
+func InTransaction(ctx context.Context, db *sql.DB, f func(*sql.Tx) error) (err error) {
+	var tx *sql.Tx
+
+	if tx, err = db.BeginTx(ctx, nil); err != nil {
+		err = lazyerrors.Error(err)
+		return
+	}
+
+	if err = f(tx); err != nil {
+		err = lazyerrors.Error(err)
+		_ = tx.Rollback()
```

Comment:
If rollback fails, it is logged right? I don't know where we configure logging for rollback on sqlite :see_no_evil:

Reply:
Afaik we don't log rollbacks in any place of our code. The error here might be returned here if transaction was already commited or rolled back, and we don't expect such situations.

Reply:
Postgres logs failed rollback, but not on go application level. If there is network issue, having log would be helpful. I _think_ sqlite _probably_ by default has similar logging enabled for things such as failed roll back. But I'm just not sure.